File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		double sumCredit = 0;
		double sumScore = 0;
		for (int i = 0; i < 20; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String subject = st.nextToken();
			double credit = Double.parseDouble(st.nextToken());
			double score = 0;
			String grade = st.nextToken();

			switch (grade) {
				case "A+":
					score = 4.5;
					break;
				case "A0":
					score = 4.0;
					break;
				case "B+":
					score = 3.5;
					break;
				case "B0":
					score = 3.0;
					break;
				case "C+":
					score = 2.5;
					break;
				case "C0":
					score = 2.0;
					break;
				case "D+":
					score = 1.5;
					break;
				case "D0":
					score = 1.0;
					break;
				case "F":
					score = 0;
					break;
			}
			if (!grade.equals("P")) {
				sumCredit += credit;
				sumScore += score * credit;
			}
		}
		System.out.println(sumScore / sumCredit);
	}
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
sumScore = 0.0
sumCredit = 0.0

for _ in range(20):
    subject, crdit, grade = input().split()
    credit = float(crdit)
    
    if grade != "P":
        if grade == "F":
            score = 0.0
        elif grade == "A+":
            score = 4.5
        elif grade == "A0":
            score = 4.0
        elif grade == "B+":
            score = 3.5
        elif grade == "B0":
            score = 3.0
        elif grade == "C+":
            score = 2.5
        elif grade == "C0":
            score = 2.0
        elif grade == "D+":
            score = 1.5
        elif grade == "D0":
            score = 1.0
            
        sumCredit += credit
        sumScore += score * credit

print(sumScore/sumCredit)